src/hammer-vlsi/par/nop.py

```diff
@@ -16,6 +16,12 @@ def fill_outputs(self) -> bool:
         self.output_ilms = []
         return True
 
+    def tool_config_prefix(self):
+        return "empty"
+
+    def version_number(self, version:str):
+        return 1
+
```

Comment:
The above are missing return type annotations and needs a space between `version` and `:str`.

Perhaps we should return a blank string or maybe change the return type to `Optional[str]` for `tool_config_prefix`?

Reply:
I think an empty string is fine.
It shouldn't really be optional, its just that mock/dummy tools still need to implement abstract methods.
I'm gonna try to create a single DummyTool that implements the abstract methods for all of these dummy tools to use.

Reply:
Now that I think about it a blank string might have issues if parts of the Hammer core are just adding ".<something>" but ".whatever" isn't going to be a valid key...

Reply:
I think if a dummy tool requires/uses config keys then it will need to override that method.